iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Branch.java

```diff
@@ -57,16 +60,79 @@
 
     Side getSide(Terminal terminal);
 
-    CurrentLimits getCurrentLimits(Side side);
+    /**
+     * @deprecated Use {@link #getOperationalLimits1(LimitType, Class)} or {@link #getOperationalLimits2(LimitType, Class)} instead.
+     */
+    @Deprecated
+    default CurrentLimits getCurrentLimits(Side side) {
+        if (side == Side.ONE) {
+            return getOperationalLimits1(LimitType.CURRENT, CurrentLimits.class);
+        } else if (side == Side.TWO) {
+            return getOperationalLimits2(LimitType.CURRENT, CurrentLimits.class);
+        }
+        throw new AssertionError("Unexpected side: " + side);
+    }
 
-    CurrentLimits getCurrentLimits1();
+    /**
+     * @deprecated Use {@link #getOperationalLimits1(LimitType, Class)} instead.
+     */
+    @Deprecated
+    default CurrentLimits getCurrentLimits1() {
+        return getOperationalLimits1(LimitType.CURRENT, CurrentLimits.class);
+    }
+
+    default List<OperationalLimits> getOperationalLimits1() {
```

Comment:
I am not sure to well understand: here we don't have the methods ````getApparentPowerLimits1()```` and ````getActivePowerLimits1()```` ?  So how do I do to get them ?

Reply:
This method will return a list of operational limit, that contains apparent limits, active power limits, and current limits.
In the API, we have to give a default behavior to the method to ensure backward compatibility. In previous IIDM version, only current limits was supported. So the implementation seems correct.